docs/imds_retry_based_on_errors.md

```diff
@@ -0,0 +1,65 @@
+# MSAL & MSI IMDS Error Handling and Retry Strategy Specification
+
+## Overview
+This document defines the error handling and retry strategy for MSAL when interacting with the IMDS (Instance Metadata Service) endpoint for Managed Identity (MSI) token acquisition.
+
+---
+
+## 1️⃣ HTTP Status Codes & Recommended Actions
+
+| **HTTP Status Code** | **Error Reason**                               | **Recommended Action**                      | **Retry Delay Strategy**                 |
+|----------------------|-----------------------------------------------|---------------------------------------------|-----------------------------------------|
+| **404**             | IMDS endpoint is updating                     | Retry with Exponential Backoff              | **1s → 2s → 4s**                        |
+| **410**             | IMDS is undergoing updates                    | Wait up to **70 seconds**, then retry      | **70s (fixed wait)**                    |
+| **429**             | IMDS Throttle limit reached                    | Retry with Exponential Backoff              | **2s → 4s → 8s → 16s**                  |
```

Comment:
Let's keep the number of times exponential backoff retries the same for all error codes that do exponential backoffs.

Reply:
Can we also define the max number of seconds for exponential backoff? 64 seconds? 32?

Reply:
added @Robbie-Microsoft

Reply:
> Let's keep the number of times exponential backoff retries the same for all error codes that do exponential backoffs.

To clarify, I was talking about 1 -> 2 -> 4... etc
Gladwin said offline: "MSI team does not want us to wait for 2 seconds in some case, as 1 second may be faster and  more beneficial"